cognite/client/_api/datapoints_subscriptions.py

```diff
@@ -185,7 +178,7 @@ def iterate_data(
         external_id: str,
         start: str | None = None,
         limit: int = DATAPOINT_SUBSCRIPTION_DATA_LIST_LIMIT_DEFAULT,
-    ) -> Iterator[DatapointSubscriptionBatch]:
+    ) -> Generator[DatapointSubscriptionBatch, None, None]:
```

Comment:
Can stay an Iterator

Reply:
I agree, but then we either need to drop `pydoclint` (you can't convince the maintainer on this one, unfortunately: https://jsh9.github.io/pydoclint/notes_generator_vs_iterator.html) or ignore all checks on return types. I think just using `Generator` is the easy solution here.